### PR TITLE
Significantly speedup protorev tests. 

### PR DESCRIPTION
**Pull request by ValarDragon**
It no longer sets up all pools in every test.

Instead tests that need pool setups have to call a separate function

Locally takes my laptop from 27 seconds to 12 seconds, we will see what happens on CI. (Was 65s)

## Summary by CodeRabbit

- **Tests**
    - Enhanced testing infrastructure for more accurate simulation of pool setups.
- **Refactor**
    - Improved setup procedures for testing, focusing on more detailed pool configurations.
- **Chores**
    - Streamlined test setup processes to better reflect real-world scenarios.

### CODE REVIEW

**Comment by ValarDragon** [comment]
Old: 65s in CI
New:
`ok  	github.com/osmosis-labs/osmosis/v24/x/protorev/keeper	19.530s`

File: x/protorev/keeper/routes_test.go [hunk 1]
```diff
@@ -129,6 +130,8 @@ func (s *KeeperTestSuite) TestBuildRoutes() {
 
 // TestBuildHighestLiquidityRoute tests the BuildHighestLiquidityRoute function
 func (s *KeeperTestSuite) TestBuildHighestLiquidityRoute() {
+	// TODO:
```
Comment (czarcas7ic) [inline x/protorev/keeper/routes_test.go:133]: Accidental TODO?

**Review by czarcas7ic — APPROVED** [review]
Nice!

File: x/protorev/keeper/routes_test.go (hunk 1 above)
Comment (ValarDragon) [reply x/protorev/keeper/routes_test.go:133]: Ah yes, I fixed what I made that todo for, ty

**Pull request merged**